File: 198-house-robber/198-house-robber.scala
object Solution {
    def rob(nums: Array[Int]): Int = {
        var a = 0
        var b = 0

        for(i <- 0 to nums.length - 1) {
            val n = nums(i)
            val c = (a + n) max b
            a = b
            b = c
        }

        a max b
    }
}

File: 198-house-robber/198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        int p1 = 0;
        int p2 = 0;
        for(int i = 0; i < nums.length; i++) {
            int n = Math.max(nums[i] + p1, p2);
            p1 = p2;
            p2 = n;
        }
        return p2;
    }
}